### DISCUSSION TITLE
Why on mineflayer we use createBot and on bedrock-protocol we use createClient?

### DISCUSSION THREAD

**Discussion by neohunter** [PrismarineJS/mineflayer#2411] [General]
The API is not suppose to be the same, or at least similar? I'm not able for example to use plugins for bedrock-protocol

```
const bedrock = require('bedrock-protocol')
const { pathfinder, Movements, goals: { GoalNear } } = require('mineflayer-pathfinder')
const bot = bedrock.createClient({
  host: 'play.lbsg.net', 
  port: 19132,         
  username: 'email@gmail.com', 
  offline: false       
})
bot.loadPlugin(pathfinder)
```

And the error is: 

TypeError: bot.loadPlugin is not a function

The same happen with prismarine-viewer

On https://github.com/PrismarineJS/bedrock-protocol/issues/116 it says that "prismarine-viewer kind of works, but needs some updates to block model mapping" but if you try to use it you get this error:

```
/Users/arnoldroa/dev/projects/aprendibot/node_modules/prismarine-viewer/lib/mineflayer.js:52
    const worldView = new WorldView(bot.world, viewDistance, bot.entity.position, socket)
                                                                        ^

TypeError: Cannot read properties of undefined (reading 'position')
    at Namespace.<anonymous> (/Users/arnoldroa/dev/projects/aprendibot/node_modules/prismarine-viewer/lib/mineflayer.js:52:73)
```

**Comment by amoraschi** [comment]
bedrock-protocol !== mineflayer

bedrock-protocol can be used to send packets to the server but it doesn't have the functions that mineflayer has. I read somewhere they wanted to combine both bedrock and java together...

bedrock-protocol == minecraft-protocol

**Reply by neohunter** [reply]
Ok, now it's more clear,  but then the question is how to make mineflayer use bedrock-protocol as client instead of minecraft-protocol.

**Reply by amoraschi** [reply]
Not implemented yet, PRs open tho

**Reply by neohunter** [reply]
whats the PR link?

**Reply by amoraschi** [reply]
The one you linked om the other discussion
